SingularityService/src/main/java/com/hubspot/singularity/resources/UserResource.java

```diff
@@ -0,0 +1,63 @@
+package com.hubspot.singularity.resources;
+
+import static com.hubspot.singularity.WebExceptions.checkBadRequest;
+
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.google.common.base.Optional;
+import com.google.inject.Inject;
+import com.hubspot.singularity.SingularityService;
+import com.hubspot.singularity.SingularityUser;
+import com.hubspot.singularity.SingularityUserSettings;
+import com.hubspot.singularity.data.UserManager;
+import com.wordnik.swagger.annotations.ApiParam;
+
+@Path(UserResource.PATH)
+@Produces({ MediaType.APPLICATION_JSON })
+public class UserResource {
+  public static final String PATH = SingularityService.API_BASE_PATH + "/users";
+
+  private final UserManager userManager;
+  private final Optional<SingularityUser> user;
+
+  @Inject
+  public UserResource(UserManager userManager,
+                      Optional<SingularityUser> user) {
+    this.userManager = userManager;
+    this.user = user;
+  }
+
+  private String getAuthUserId() {
+    checkBadRequest(user.isPresent() && user.get().getName().isPresent(), "Singularity userId must be provided by auth");
+    return user.get().getName().get();
+  }
```

Comment:
This method needs some work:
- The error message isn't accurate because we're not using the `userId` query param anymore.
- Being unauthenticated should return an Unauthorized response, not a Bad Request response. Use SingularityAuthorizationHelper as a guide here; it uses `checkUnauthorized(user.isPresent(), "Please log in to perform this action.");` for stuff like this.
- It's using the `name` field of SingularityUser (`Thomas Petr`) when it should be using the `id` field (`tpetr`).